### PR TITLE
fix(TPRUN-5912): fixed credential flow for reactive app

### PR DESCRIPTION
**Pull request by lmeunier-talend**
**What is the problem this Pull Request is trying to solve?**
 
**What is the chosen solution to this problem?**

- Fixed the `TalendJwtConverter` for crendentials flow in the reactive lib
 
**Link to the JIRA issue**

 
**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

File: daikon-spring/daikon-spring-reactive/daikon-spring-reactive-auth/src/main/java/org/talend/daikon/spring/reactive/sat/authentication/TalendJwtConverter.java [hunk 1]
```diff
@@ -21,10 +21,10 @@ public class TalendJwtConverter implements Converter<Jwt, Collection<GrantedAuth
 
     public static final String TALEND_SAT_PERMISSIONS_FIELD = "https://talend.cloud/permissions";
 
+    @SuppressWarnings("unchecked")
     public Collection<GrantedAuthority> convert(Jwt jwt) {
         List<String> ents = Arrays.asList(((String) jwt.getClaims().getOrDefault(TALEND_ENTITLEMENTS_FIELD, "")).split(","));
-        List<String> permsList = ((JSONArray) jwt.getClaims().getOrDefault(TALEND_PERMISSIONS_FIELD, new JSONArray())).stream()
-                .map(Object::toString).toList();
+        List<String> permsList = (List<String>) jwt.getClaims().getOrDefault(TALEND_PERMISSIONS_FIELD, Collections.emptyList());
         List<String> satPerms = ((JSONArray) jwt.getClaims().getOrDefault(TALEND_SAT_PERMISSIONS_FIELD, new JSONArray())).stream()
```
Comment (jehervy) [inline daikon-spring/daikon-spring-reactive/daikon-spring-reactive-auth/src/main/java/org/talend/daikon/spring/reactive/sat/authentication/TalendJwtConverter.java:28]:
```suggestion
        List<String> satPerms = (List<String>) jwt.getClaims().getOrDefault(TALEND_SAT_PERMISSIONS_FIELD, Collections.emptyList());
```
Shouldn't we apply the same change for SAT permissions?

File: daikon-spring/daikon-spring-reactive/daikon-spring-reactive-auth/src/main/java/org/talend/daikon/spring/reactive/sat/authentication/TalendJwtConverter.java [hunk 2]
```diff
@@ -21,10 +21,10 @@ public class TalendJwtConverter implements Converter<Jwt, Collection<GrantedAuth
...
                 .map(Object::toString).toList();
```
Comment (jehervy) [inline daikon-spring/daikon-spring-reactive/daikon-spring-reactive-auth/src/main/java/org/talend/daikon/spring/reactive/sat/authentication/TalendJwtConverter.java:29]:
```suggestion
        List<String> satPerms = (List<String>) jwt.getClaims().getOrDefault(TALEND_SAT_PERMISSIONS_FIELD, Collections.emptyList());
```
Shouldn't we apply the same change for SAT permissions?

Comment (lmeunier-talend) [reply daikon-spring/daikon-spring-reactive/daikon-spring-reactive-auth/src/main/java/org/talend/daikon/spring/reactive/sat/authentication/TalendJwtConverter.java:29]: i was wondering about it too

Comment (lmeunier-talend) [reply daikon-spring/daikon-spring-reactive/daikon-spring-reactive-auth/src/main/java/org/talend/daikon/spring/reactive/sat/authentication/TalendJwtConverter.java]: done

**Review by jehervy — APPROVED** [review]
LGTM

**Review by jehervy — APPROVED** [review]

**Pull request merged**